ts/src/gate.ts

```diff
@@ -5796,7 +5796,7 @@ export default class gate extends Exchange {
         return tiers;
     }
 
-    parseMarketLeverageTiers (info, market: Market = undefined): LeverageTier[] {
+    parseMarketLeverageTiers (info: Dict, market: Market = undefined): LeverageTier[] {
```

Comment:
![image](https://github.com/ccxt/ccxt/assets/43336371/588e7b74-097e-40b6-bd71-c309dce17c16)
it seems we might get an array here

Reply:
I think this is a situation where we need something like [this type](https://github.com/ccxt/ccxt/pull/22387), without it, we'll need to rewrite `parseLeverageTiers` for a lot of exchanges because it might pass a dictionary or a list to `parseMarketLeverageTiers`. Right now the transpiler can have a type of `Dict | List` for arguments